pkg/domain/infra/abi/network.go

```diff
@@ -144,6 +171,33 @@ func (ic *ContainerEngine) NetworkExists(ctx context.Context, networkname string
 
 // Network prune removes unused cni networks
 func (ic *ContainerEngine) NetworkPrune(ctx context.Context, options entities.NetworkPruneOptions) ([]*entities.NetworkPruneReport, error) {
+	// get all filters
+	filters, err := netutil.GenerateNetworkPruneFilters(options.Filters)
+	if err != nil {
+		return nil, err
+	}
+	danglingFilterFunc, err := ic.createDanglingFilterFunc(true)
+	if err != nil {
+		return nil, err
+	}
+	filters = append(filters, danglingFilterFunc)
+	nets, err := ic.Libpod.Network().NetworkList(filters...)
+	if err != nil {
+		return nil, err
+	}
+
+	pruneReport := make([]*entities.NetworkPruneReport, 0, len(nets))
+	for _, net := range nets {
+		pruneReport = append(pruneReport, &entities.NetworkPruneReport{
+			Name:  net.Name,
+			Error: ic.Libpod.Network().NetworkRemove(net.Name),
+		})
+	}
+	return pruneReport, nil
+}
+
+// danglingFilter function is special and not implemented in libnetwork filters
+func (ic *ContainerEngine) createDanglingFilterFunc(wantDangling bool) (types.FilterFunc, error) {
```

Comment:
Why is wantDangling an argument to this function? If we do not want dangling we should just not call this function.

Reply:
My understanding was that this filter introduces a tristate:
- no filter -> list any network, dangling or not
- dangling=true -> list only dangling networks
- dangling=false -> list only non-dangling networks

For this, I thought of passing the `wantDangling` variable to invert the selection when needed.

Reply:
yeah you are right